star-rx72n-firmware/gpio_test/host/gpio_verify.py

```
@@ -382,18 +572,30 @@ def main() -> None:
+    parser.add_argument("--auto-map", action="store_true",
+                        help="Auto-discover which channel is wired to which "
+                             "pin. Captures 2 cycles and prints a paste-ready "
+                             "CHANNEL_TO_PIN dict. Skip verification step.")
     args = parser.parse_args()
 
+    # Auto-map forces 2 capture cycles so the cycle-gap anchor is always in
+    # range regardless of when the capture starts relative to firmware boot.
+    if args.auto_map and args.cycles < 2:
+        args.cycles = 2
+
     print(f"GPIO Breakout Board Verification")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)